src/transformers/modeling_utils.py

```diff
@@ -1282,6 +1292,18 @@ def load(module: nn.Module, prefix=""):
 
         return model
 
+    def retrieve_modules_from_weights(self, weight_keys):
```

Comment:
This function is important so that we can run the model specific `_init_weights(...)` function on each module. It's a bit dangerous though since there is a chance that weights that shouldn't be overwritten by random weights can be overwritten in weird edge cases. => to remedy this we can add a check though